Server/controllers/user.controller.js

```diff
@@ -1,17 +1,66 @@
-"use strict";
+const datalayerFactory = require('../datalayer/factory.datalayer');
+const userDataLayer = require('../datalayer/user.datalayer');
 /**
  * Class representing users controller
  *
  * @class usersController
  */
 class usersController {
 
-  static createUser (req, res) {}
+  static async createUser (req, res) {
+    let reqObject = req.body;
+        try {
+            await userDataLayer.createUser(reqObject);
+        } catch(err) {
+            res.status(400).send(err.message);
+            return;
+        }
+        res.status(200);
+        res.send("User created successfully");
+  }
 
-  static getUsersByUserId (req, res) {}
 
-  static updateUserById (req, res) {}
+  static async getUsersByUserId (req, res) {
+    let userid = req.params._id;
+    let user;
+    try {
+      user = await userDataLayer.getUsersByUserId(userid);
+    }
+    catch(err) {
+      res.status(400).send(err.message);
+      return;
+    }
+    res.status(200).json(user);
+  }
 
-  static deleteUserById (req, res) {}
+
+  static async updateUserById (req, res) {
+    let reqbody = req.body;
+    let userid = req.params._id;
+    try {
+      await userDataLayer.updateUserById(userid , reqbody);
```

Comment:
We shouldn't pass request body directly. Instead extract the parameters and create a new object and pass that. 
Also check for existence of paramters etc.

Reply:
New object created as suggested.